### DISCUSSION TITLE
A question about IfcAdvancedBrep

### DISCUSSION THREAD

**Discussion by bimintelligence** [IfcOpenShell/IfcOpenShell#2733] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/73410751/216350608-22ac7cdf-179d-44a0-a68d-946b397cdced.png)

**Comment by aothms** [comment, marked as answer]
1) it's a polyline in the UV space of the IfcCylindricalSurface

2) Have a look at https://github.com/buildingSMART/Sample-Test-Files/tree/master/IFC%204.0/NURBS/Bentley%20Building%20Designer/Cylinder and please report back your findings

**Reply by bimintelligence** [reply]
Hi, aothms. This example is described below, using IfcIndexedPolyCurve instead of IfcPolyline.
I'm a little confused.
![image](https://user-images.githubusercontent.com/73410751/216510665-956b44b9-4bc2-4e8d-99d3-edc99d8a7c99.png)

**Reply by aothms** [reply]
They don't have the parametric representation of the curve in UV, just a indexed poly curve using arc segments (the poly-curve despite is name is not necessarily polygonal...) Also here, I'm not a fan of poly-curve for edge curves, because it maps more to a loop then an individual edge.

It's just an example, so I guess then there really isn't an online example of this in IFC.

I have two suggestions:

- Ping @TLiebich (who has [judging from the style] created the diagram). Maybe he knows if at that time there was also a sample model.
- Look for plain step examples, the chances of finding something there will be much higher. If you can't find anything online, you can use OpenCascade (or pythonOCC). Basically just use `BRepPrimAPI_MakeCylinder` and `STEPControl_Writer`. By default it should include surface curves (`write.surfacecurve.mode` defaults to `on`)

**Reply by bimintelligence** [reply]
Thanks aothms, your answer inspired me a lot.
